### DISCUSSION TITLE
NestJS support

### DISCUSSION THREAD

**Discussion by starskyreverie** [scalar/scalar#698] [Ideas]
Would appreciate support for NestJS. I know it's built on top of Express / Fastify and there are workarounds to include the middleware, but explicit documentation on it would be great.

**Comment by marclave** [comment]
hey @pq43 , i have fantastic news! it's on our roadmap for Q1 of next year :) Hopefully the earlier part of Q1

**Reply by starskyreverie** [reply]
legend, thank you

**Reply by marclave** [reply]
why leave for tomorrow what can be done today :)

here's the PR for it :) https://github.com/scalar/scalar/pull/700

let me know if you have any feedback in the PR! I ended up using the bare `app.use` instead of the class based approach for starters, something we can fast follow later ✨

**Reply by starskyreverie** [reply]
wow, that is really amazing. we ended up using `app.use` internally too :)

thank you so much!

**Reply by marclave** [reply]
cool :) 

I'll drop a comment here when it's released! Also if you use discord I will update there on package release too!

https://discord.com/invite/8HeZcRGPFS

**Reply by hanspagel** [reply]
And released :)

https://www.npmjs.com/package/@scalar/nestjs-api-reference
